controlalgorithm/persistent_data.py

```diff
@@ -0,0 +1,132 @@
+"""
+Date: Feb 20, 2020
+Author: Sam Wu
+Contents: Store persistent data from API calls
+Location (lat/lon)
+Cloud Cover Percentage
+External Temp
+"""
+
+import datetime
+import dotenv
+import json
+import os
+import pandas
+import pvlib.solarposition
+import requests
+import sys
+
+import user_defined_exceptions as exceptions
+
+"""
+API Keys and Endpoints
+"""
+dotenv.load_dotenv()
+
+OPENCAGE_API_KEY = os.getenv("OPENCAGE_API_KEY")
+OPENCAGE_URL = "https://api.opencagedata.com/geocode/v1/json?key={OPENCAGE_API_KEY}&q={place_name}&pretty=1"
+
+DARKSKY_API_KEY = os.getenv("DARKSKY_API_KEY")
+DARKSKY_URL = "https://api.darksky.net/forecast/{DARKSKY_API_KEY}/{lat},{lon}"
+
+"""
+Forward Geocoding
+Given a location or place name, 
+get the latitude/longitude
+and time zone of the place
+Then put the values in persistent_data.json
+"""
+def get_lat_lon():
+    # first check if json data file is present and readable
+    if os.path.isfile("persistent_data.json") and os.access("persistent_data.json", os.R_OK):
+        with open("persistent_data.json", "r+") as fp:
+            persistent_data_dict = json.load(fp)
+        lat = persistent_data_dict["lat"]
+        lon = persistent_data_dict["lon"]
+        timezone_adjustment = persistent_data_dict["timezone_adjustment"]
+
+    # if not, get the data and create the json file
+    else:
+        persistent_data_dict = dict()
+
+        place_name = input("Enter location or place name: ")
```

Comment:
We can't wait for user input on the RPI code. The RPI code runs entirely headlessly. We should find another way to get the user's location. Or we can just hardcode it for now.

Reply:
Ideally in our system, the only thing that really interacts with the user is the web app. So we either use an API end point to send configuration (ex. timezone) over the very first time the blind is paired or we find some other method that doesn't involve direct user input in the python console.

Reply:
In a production setting, the external application should send this data. @kdehaan 
For the demo, we can hard code this for convenience and time constraints.